development/workflow.rst

```diff
@@ -0,0 +1,667 @@
+Before serious coding effort is committed to a ticket, the developer should either undertake an informal design review while creating the JIRA story, or more formally use the :abbr:`RFC (Request for Comment)` and :abbr:`RFD (Request for Discussion)` processes for key design decisions.
+
+.. TODO: link to RFC/RFC process doc
+
+.. _workflow-review-assign:
+
+Assign a reviewer
+-----------------
+
+On your ticket's JIRA page, use the **Workflow** button to switch the ticket's state to **In Review**.
+JIRA will ask you to assign reviewers.
+
+In your JIRA message requesting review, indicate how involved the review work will be ("quick" or "not quick").
+The reviewer should promptly acknowledge the request, indicate whether they can do the review, and give a timeline for when they will be able to accomplish the request.
+This allows the developer to seek an alternate reviewer if necessary.
+
+Any team member in Data Management can review code; it is perfectly fine to draw reviewers from any segment of DM.
+For major changes, it is good to choose someone more experienced than yourself.
+For minor changes, it may be good to choose someone less experienced than yourself.
+For large changes, more than one reviewer may be assigned, possibly split by area of the code.
+
+Code reviews performed by peers are useful for a number of reasons:
+
+- Peers are a good proxy for maintainability.
+- It's useful for everyone to be familiar with other parts of the system.
+- Good practices can be spread; bad practices can be deprecated.
+
+All developers are expected to make time to perform reviews.
+The System Architect can intervene, however, if a developer is overburdened with review responsibility.
+
+.. _workflow-code-review-process:
+
+Code review discussion
+----------------------
+
+Code review discussion should happen on the GitHub pull request, with the reviewer giving a discussion summary and conclusive thumbs-up on the JIRA ticket.
+
+GitHub pull requests are ideal venues for pull request discussion since individual commit diffs can be annotated and referenced.
+Be sure to make comments only from the **Conversation** and **Files changed** tabs---*not the Commits tab*.
+Any comments on code patches from the Commits tab will be lost if the developer amends and force pushes commits to the pull request.
+
+Code reviews are a collaborative check-and-improve process.
+Reviewers do not hold absolute authority, nor can developers ignore the reviewer's suggestions.
+The aim is to discuss, iterate, and improve the pull request until the work is ready to be deployed on ``master``.
+
+If the review becomes stuck on a design decision, that aspect of the review can be elevated into an RFC to seek team-wide consensus.
+
+If an issue is outside the ticket's scope, the reviewer should file a new ticket.
+
+Once the iterative review process is complete, the reviewer should switch the JIRA ticket's state to **Reviewed**.
+This indicates the ticket is ready to be merged.
+
```

Comment:
If there are multiple reviewers, the convention is for each reviewer to remove his/her name from the Reviewers list; the last one switches the status to "Reviewed".